leapp/utils/report.py

```diff
@@ -100,8 +100,17 @@ def generate_report_file(messages_to_report, context, path):
                 remediation = Remediation.from_dict(message.get('detail', {}))
                 if remediation:
                     f.write('Remediation: {}\n'.format(remediation))
-                f.write('Key: {}\n'.format(message['key']))
+                if not report_schema or report_schema > '1.0.0':
```

Comment:
@fernflower  wdyt about transform it to tuple? `(1, 0, 0)`? Just thinking about that. This will be ok so far we have just numbers up-to-9. however in case we will have a version "1.10.0", such comparison will not be the good way. It's not problem right now, this should work fine so far. But thinking whether wouldn't be better to parse it to a better data format already to not have to think about that in future.

Reply:
Perfect, will do

Reply:
@fernflower  Thanks. Looks good to me. Btw, in case you would like to do it using the callback functions for the `action` argument, the solution could looks something like that:

```python
#!/usr/bin/python3

import argparse

class VersionParserAction(argparse.Action):
    def __init__(self, option_strings, dest, **kwargs):
        super(VersionParserAction, self).__init__(option_strings, dest, **kwargs)
    def __call__(self, parser, namespace, version_string, option_string=None):
        parsed_version = tuple([int(i) for i in version_string.split('.')])
        setattr(namespace, self.dest, parsed_version)

parser = argparse.ArgumentParser()
parser.add_argument(
    '--report-schema', help='Specify report schema version for leapp-report.json',
    choices=['1.0.0', '1.1.0'], default=(1, 1, 0),
    action=VersionParserAction
)

### for tests tests
print(parser.parse_args('--report-schema 1.1.0'.split()))
print(parser.parse_args('--report-schema 1.0.0'.split()))

args = parser.parse_args('--report-schema 1.1.0'.split())
print(args.report_schema)
args = parser.parse_args(''.split())
print(args.report_schema)
```

Reply:
tbh I prefer my version and tuple conversion before actual usage.

Reply:
no problem.